src/nuclide.cpp

```diff
@@ -404,7 +497,245 @@ double Nuclide::elastic_xs_0K(double E) const
   return (1.0 - f)*elastic_0K_[i_grid] + f*elastic_0K_[i_grid + 1];
 }
 
-void Nuclide::calculate_urr_xs(int i_temp, double E)
+void Nuclide::calculate_xs(int i_sab, double E, int i_log_union,
+  double sqrtkT, double sab_frac)
+{
+  auto& micro_xs = simulation::micro_xs[i_nuclide_];
+
+  // Initialize cached cross sections to zero
+  micro_xs.elastic = CACHE_INVALID;
+  micro_xs.thermal = 0.0;
+  micro_xs.thermal_elastic = 0.0;
+
+  // Check to see if there is multipole data present at this energy
+  bool use_mp = false;
+  if (multipole_) {
+    use_mp = (E >= multipole_->E_min_ && E <= multipole_->E_max_);
+  }
+
+  int i_temp = -1;
+
+  // Evaluate multipole or interpolate
+  if (use_mp) {
+    // Call multipole kernel
+    double sig_s, sig_a, sig_f;
+    std::tie(sig_s, sig_a, sig_f) = multipole_->evaluate(E, sqrtkT);
+
+    micro_xs.total = sig_s + sig_a;
+    micro_xs.elastic = sig_s;
+    micro_xs.absorption = sig_a;
+    micro_xs.fission = sig_f;
+    micro_xs.nu_fission = fissionable_ ?
+      sig_f * this->nu(E, EmissionMode::total) : 0.0;
+
+    if (simulation::need_depletion_rx) {
+      // Only non-zero reaction is (n,gamma)
+      micro_xs.reaction[0] = sig_a - sig_f;
+
+      // Set all other reaction cross sections to zero
+      for (int i = 1; i < DEPLETION_RX.size(); ++i) {
+        micro_xs.reaction[i] = 0.0;
+      }
+    }
+
+    // Ensure these values are set
+    // Note, the only time either is used is in one of 4 places:
+    // 1. physics.cpp - scatter - For inelastic scatter.
+    // 2. physics.cpp - sample_fission - For partial fissions.
+    // 3. tally.F90 - score_general - For tallying on MTxxx reactions.
+    // 4. nuclide.cpp - calculate_urr_xs - For unresolved purposes.
+    // It is worth noting that none of these occur in the resolved
+    // resonance range, so the value here does not matter.  index_temp is
+    // set to -1 to force a segfault in case a developer messes up and tries
+    // to use it with multipole.
+    micro_xs.index_temp = -1;
+    micro_xs.index_grid = 0;
+    micro_xs.interp_factor = 0.0;
+
+  } else {
+    // Find the appropriate temperature index.
+    double kT = sqrtkT*sqrtkT;
+    double f;
+    switch (settings::temperature_method) {
+    case TEMPERATURE_NEAREST:
+      {
+        double max_diff = INFTY;
+        for (int t = 0; t < kTs_.size(); ++t) {
+          if (std::abs(kTs_[t] - kT) < max_diff) {
+            i_temp = t;
```

Comment:
Am I reading this correctly?  It looks like this will always set `i_temp = 0`

Reply:
You're right -- there's something missing here. I'll fix it. Thanks!